src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1157,11 +1157,15 @@ protected virtual void AssignImpl(BoundNode node, BoundExpression value, RefKind
             {
                 case BoundKind.DeclarationPattern:
                     {
-                        var local = (BoundDeclarationPattern)node;
-                        LocalSymbol symbol = local.LocalSymbol;
-                        int slot = GetOrCreateSlot(symbol);
-                        SetSlotState(slot, assigned: written || !this.State.Reachable);
-                        if (written) NoteWrite(symbol, value, read);
+                        var pattern = (BoundDeclarationPattern)node;
+                        Symbol symbol = pattern.Variable as LocalSymbol;
+                        if (symbol != null)
```

Comment:
Is it useful to cast to object here for null check?
Does the case were pattern.Variable is a field need to be handled too? If not, consider adding a comment.